reference/upgrade-guide.md

```diff
@@ -0,0 +1,162 @@
+the Language Guide.
+
+::: tip
+Changes are grouped in terms of their **impact**. The word "impact" here is a
+combination of "how important is the change" along with "how many existing WDL
+documents are likely to be affected by this change". They are subjective
+determinations made by the upgrade guide curator during the curator process, but
+they are often helpful in quickly scanning upgrade elements to see which might
+apply to your situation.
+:::
+
+# WDL v1.2
+
+WDL v1.2 introduces a great deal of new syntax and changes to the specification.
+Notably, changes were backwards compatible with all previous `v1.x` releases.
+This sections only covers the high points related to upgrading—the full release
+notes can be found [here][wdl-v1.2-release-notes].
+
+### Checklist
+
+Use this checklist to ensure you hit all of the sections.
+
+#### Moderate impact changes 
+
+<input type="checkbox" /> New `Directory` type ([link](#new-directory-type)).<br />
+<input type="checkbox" /> Replacement of `runtime` section with `requirements` and `hints` sections ([link](#replacement-of-runtime-with-requirements-and-hints)). <br />
+<input type="checkbox" /> New standard library functions
+([link](#new-standard-library-functions)). <br />
+<input type="checkbox" /> Multi-line strings ([link](#multi-line-strings)). <br />
+<input type="checkbox" /> Optional `input:` statement ([link](#optional-input-statement)). <br />
+
+#### Low impact changes 
+
+<input type="checkbox" /> Addition of workflow `hints` ([link](#addition-of-workflow-hints)). <br />
+<input type="checkbox" /> New requirements and hints keys ([link](#new-requirements-and-hints-keys)). <br />
+<input type="checkbox" /> Metadata sections for structs ([link](#metadata-sections-for-structs)). <br />
+<input type="checkbox" /> Exponentiation operator ([link](#exponentiation-operator)). <br />
+
+## Moderate impact changes
+
+### New `Directory` type
+
+The `Directory` type was introduced in
+[#641](https://github.com/openwdl/wdl/pull/641) to better semantically indicate
+the use of a directory. If the intention of any of your arguments is to be used
+to refer to a directory on the filesystem, you are encouraged to update the
+parameters to a `Directory` type.
+
+### Replacement of section `runtime` with `requirements` and `hints` sections
+
+The `runtime` section, which previously held both requirement contraints and
+hints to the execution engine, has now been split into the `requirements`
+([#540](https://github.com/openwdl/wdl/issues/540)) section and `hints`
+([#541](https://github.com/openwdl/wdl/issues/541)) section respectively. You
+should split out these keys based on the definitions in the
+specification<sup>[1](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#-requirements-section),
+[2](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#-hints-section)</sup>.
+
+🗑️ This change deprecates the `runtime` section, which will be removed in WDL
```

Comment:
Maybe add something like: You can continue to use the runtime section in your wdl documents without any changes